csob/resources/payment/init.py

```diff
@@ -0,0 +1,100 @@
+            raise ValueError('payOperation invalid value')
+        local_json['payOperation'] = pay_operation
+
+        if pay_method != 'card':
+            raise ValueError('payMethod invalid value')
+        local_json['payMethod'] = pay_method
+
+        if type(total_amount) != int:
+            raise ValueError('totalAmount invalid value')
+        local_json['totalAmount'] = total_amount
+
+        if currency not in ['CZK', 'EUR', 'USD', 'GBP', 'HUF', 'PLN', 'HRK', 'RON', 'NOK', 'SEK']:
+            raise ValueError('curreny invalid value')
+        local_json['currency'] = currency
+
+        if type(close_payment) != bool:
+            raise ValueError('closePayment invalid value')
+        local_json['closePayment'] = close_payment
+
+        if len(return_url) > 300:
+            raise ValueError('returnUrl is too long')
+        local_json['returnUrl'] = return_url
+
+        if return_method not in ['POST', 'GET']:
+            raise ValueError('returnMethod invalid value')
+        local_json['returnMethod'] = return_method
+
+        if len(description) > 255:
+            raise ValueError('description is too long')
+        local_json['description'] = description
+
+        if language not in ['CZ', 'EN', 'DE', 'FR', 'HU', 'IT', 'JP', 'PL', 'PT', 'RO', 'RU', 'SK', 'ES', 'TR', 'VN',
+                            'HR', 'SI']:
+            raise ValueError('language is invalid')
+        local_json['language'] = language
+
+        if merchant_data is not None:
+            if len(merchant_data) > 255:
+                raise ValueError('merchantData is too long')
+            local_json['merchantData'] = merchant_data
+
+        if customer_id is not None:
+            if len(customer_id) > 50:
+                raise ValueError('customerId is too long')
+            local_json['customerId'] = customer_id
+
+        if cart is None:
+            cart = [{
+                'name': 'Your purchase',
+                'quantity': 1,
+                'amount': total_amount,
+            }]
+        local_json['cart'] = cart
+
+        if ttl_sec is not None:
+            if type(ttl_sec) == int and 300 <= ttl_sec <= 1800:
+                local_json['ttlSec'] = ttl_sec
+            else:
+                raise ValueError('ttlSec not in range')
+
+        if logo_version is not None:
+            if type(logo_version) == int:
+                local_json['logoVersion'] = logo_version
+            else:
+                raise ValueError('logoVersion is not a number.')
+
+        if color_scheme_version is not None:
+            if type(color_scheme_version) == int:
+                local_json['colorSchemeVersion'] = color_scheme_version
+            else:
+                raise ValueError('colorSchemeVersion is not a number.')
+
+        local_json["signature"] = self.get_signature(key, local_json)
+        return self.parse_response(self.session.post(self.get_url(), data=json.dumps(local_json)))
```

Comment:
Tenhle řádek se dost opakuje... nebylo by lepší, kdyby ty metody vracely jen data pro request a to parse response by volal parent? Něco jako:
```
def post(self):
   self.parse_response(selfsession.post(self.get_url(), data=self.get_post_data())
```

Reply:
Massivně refactorováno